lib/src/types/hd_key_pair.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:cryptography/cryptography.dart';
+import 'package:ed25519_hd_key/ed25519_hd_key.dart';
+
+class HDKeyPair implements SimpleKeyPair {
+  HDKeyPair._({
+    required KeyData private,
+    required List<int> publicKey,
+  })   : _private = private,
+        _publicKey = publicKey;
+
+  static Future<KeyPair> fromSeed(List<int> seed, String hdPath) async {
+    final KeyData private = await ED25519_HD_KEY.derivePath(hdPath, seed);
+    final List<int> publicKey = await ED25519_HD_KEY.getPublicKey(
+      private.key,
+      false,
+    );
+    return HDKeyPair._(private: private, publicKey: publicKey);
+  }
+
+  final KeyData _private;
+  final List<int> _publicKey;
+
+  @override
+  Future<SimpleKeyPairData> extract() async => Future<SimpleKeyPairData>.value(
+        SimpleKeyPairData(
+          _private.key,
+          publicKey: await extractPublicKey(),
+          type: KeyPairType.ed25519,
+        ),
+      );
```

Comment:
No need to use `Future.value()` here, the value itself can be returned:

```suggestion  @override
  Future<SimpleKeyPairData> extract() async => SimpleKeyPairData(
        _private.key,
        publicKey: await extractPublicKey(),
        type: KeyPairType.ed25519,
      );
```

Same for the next methods here (just add async to the methods).

Reply:
It depends, without the async it is needed and that was the intention! because this function is not per-se async. Thanks for pointing it out!

Reply:
In this case there's an `await` so you are 100% correct. Thanks.

Reply:
Any function returning `Future` is async, so next methods can be updated as well. E.g.:

```dart
Future<List<int>> extractPrivateKeyBytes() async => _private.key;
```

is just the shorter version of:

```dart
Future<List<int>> extractPrivateKeyBytes() =>
      Future<List<int>>.value(_private.key);
```

Reply:
Actually the default crypto package had very similar code so I guess it's not so important.

Reply:
Definitely not important, just looks a little bit more readable to me.